src/rgw/rgw_rest_pubsub.cc

```diff
@@ -361,7 +361,7 @@ void RGWPSCreateTopicOp::execute(optional_yield y) {
   // master request will replicate the topic creation.
   if (!driver->is_meta_master()) {
     op_ret = rgw_forward_request_to_master(
-        this, *s->penv.site, s->user->get_id(), &bl_post_body, nullptr, s->info, y);
+        this, *s->penv.site, s->owner.id, &bl_post_body, nullptr, s->info, y);
```

Comment:
could a users have different permission in different zones?

Reply:
so, why not passing the user and doing the validation on the master zone as well?

Reply:
these requests are forwarded as the multisite system user, which is overriding permission checks anyway

passing `s->owner.id` (which represents either a `rgw_user` or `rgw_account_id`) here is necessary to make sure that the master zone sees `s->auth.identity->get_account()` for forwarded requests from account users

without this change, the multisite test case `test_topic_notification_sync` fails with an account user because the forwarded CreateTopic request ends up being owned by the `s->user` and not their account

Reply:
(the accounts pr made the same change to all rgw_forward_request_to_master() calls in rgw_op.cc)